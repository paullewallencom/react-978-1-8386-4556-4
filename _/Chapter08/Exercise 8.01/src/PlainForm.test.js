import React from 'react';
import {render, fireEvent, wait} from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import {PlainForm} from './PlainForm';

test('renders correctly', () => {
  const {container} = render(<PlainForm/>);
  expect(container).toMatchSnapshot();
});

test('loads and displays form', async () => {
  //Arrange
  const { container } = render(<PlainForm />);
  const email = container.querySelector('input[type="text"]');
  const password = container.querySelector('input[type="password"]');

  //Act
  await wait(() => {
    fireEvent.change(email, {
      target: {
        value: 'example@email.com'
      }
    });
    fireEvent.change(password, {
      target: {
        value: 'password'
      }
    })
  });


  //Assert
  expect(email.value).toBe('example@email.com');
  expect(password.value).toBe('password');
});
