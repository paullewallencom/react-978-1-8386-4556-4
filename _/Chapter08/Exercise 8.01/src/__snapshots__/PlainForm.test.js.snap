// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <form
    action="/login.php"
    novalidate=""
  >
    <label>
      Email:
      <input
        type="text"
      />
    </label>
    <label>
      Password:
      <input
        type="password"
      />
    </label>
    <input
      type="submit"
      value="Login"
    />
  </form>
</div>
`;
